File: Java/demos/desktop/libraries/bmp280/src/main/java/io/treehopper/desktop/demos/bmp280.java
package io.treehopper.desktop.demos;

import io.treehopper.TreehopperUsb;
import io.treehopper.desktop.ConnectionService;
import io.treehopper.libraries.sensors.pressure.bmp280.Bme280;
import io.treehopper.libraries.sensors.pressure.bmp280.Bmp280;

import java.util.List;

public class bmp280 {
    public static void main(String[] args) throws InterruptedException {
        TreehopperUsb board = ConnectionService.getInstance().getBoards().get(0);
        board.connect();

        List<Bmp280> sensors;
        while(true) {
            sensors = Bmp280.Probe(board.i2c, true);
            if(sensors.isEmpty()) {
                System.out.println("No BMP280 or BME280 found.");
                Thread.sleep(500);
            } else {
                break;
            }
        }

        Bmp280 sensor = sensors.get(0);
        sensor.setAutoUpdateWhenPropertyRead(false);

        while(true)
        {
            sensor.update();
            System.out.println("Altitude: " + sensor.getAltitude() + "m");
            Thread.sleep(100);
        }
    }
}

File: Python/library_examples/sensors/pressure/bmp280.py
from time import sleep

from treehopper.api import *
from treehopper.libraries.sensors.pressure import Bmp280, Bme280

sensor = None
board = find_boards()[0]
board.connect()
devices = Bme280.probe(board.i2c)
if len(devices) > 0:
    sensor = devices[0]
else:
    raise RuntimeError("No BMP280 was found.")

sensor.auto_update_when_property_read = False

while board.connected:
    sensor.update()
    print(f"Pressure: {sensor.atm} Atm")
    print(f"Temperature: {sensor.celsius} °C ({sensor.fahrenheit} °F)")
    print(f"Altitude: {sensor.altitude} m")
    print(f"Humidity: {sensor.relative_humidity}%")
    sleep(0.5)
